models/rewards_validators/validators/orders/simcard/HasActivityOnSimcardValidator.php

```diff
@@ -131,6 +135,133 @@ public function getMinCountDateForStart():int {
+				$pairs[] = [
+					'status' => $status->id,
+					'ban' => $connection->relatedContracts->ban,
+					'phone' => $connection->ctn,
+					'dmpRow' => null
+				];
+			}
+		}
+		$dmpRows = Yii::$container->get(DimSubscriberAr::class)->findActivePhoneBatch(new BanWithPhoneCollection($pairs));
+
+		foreach ($pairs as &$pair) {
+			foreach ($dmpRows as $row) {
+				if ($row->isEqualPhone($pair['phone']) && $row->isEqualBan($pair['ban'])) {
+					$pair['dmpRow'] = $row;
+				}
+			}
+		}
+		unset($pair);
+
+		foreach ($collection as $statusRow) {
+			foreach ($pairs as $pair) {
+				if ($statusRow->id === $pair['status'] && null === $pair['dmpRow']) {
+					$statusRow->status = RewardBonusValidatorInterface::STATUS_DONE_FAIL;
+				}
+			}
+		}
+
+		$activePairs = array_filter($pairs, static fn(array $pair) => null !== $pair['dmpRow']);
+		$callsDuration = Yii::$container->get(FctUsagePrepChaNAr::class)->getSumDurationCallBatch(
+			FctUsagePrepChaN::CALL_DIRECTION_INCOMING, $activePairs, $this->countDaysForCalls
+		);
+
+		$withOutCalls = [];
+		foreach ($activePairs as $pair) {
+			foreach ($callsDuration as $dmpRow) {
+				$model = new DimSubscriberAr([
+					'subs_key' => $dmpRow['subs_key'],
+					'ban_key' => $dmpRow['ban_key']
+				]);
+				if ($model->isEqualPhone($pair['phone']) && $model->isEqualBan($pair['ban'])) {
+					if (($dmpRow['sum_'] * 60) < 1) {
+						$withOutCalls[] = $pair;
+					}
+				} else {
+					$withOutCalls[] = $pair;
+				}
+			}
+		}
+
+		$postpaid = array_filter($activePairs, static fn(array $item) => $item['dmpRow']->isPostpaid());
+		$prepaid = array_filter($activePairs, static fn(array $item) => $item['dmpRow']->isPrepaid());
+
+		$feeWithPositivePostpaid = [];
+		if (count($postpaid) > 0) {
+			$fee = Yii::$container->get(FctChargeAr::class)->getSumRegularByBanBatch(
+				new BanWithPhoneCollection(array_map(static function(array $item) {
+					return [
+						'phone' => $item['phone'],
+						'ban' => $item['ban']
+					];
+				}, $postpaid))
+			);
+			$feeWithPositivePostpaid = array_filter($fee, static fn(array $item) => $item['total_sum'] > 0);
+		}
+
+		$feeWithPositivePrepaid = [];
+		if (count($prepaid) > 0) {
+			$feePrepaid = Yii::$container->get(FctPrepChargesNAr::class)->getSumFeeBatch(
+				new BanWithPhoneCollection(array_map(static function(array $item) {
+					return [
+						'phone' => $item['phone'],
+						'ban' => $item['ban']
+					];
+				}, $prepaid))
+			);
+			$feeWithPositivePrepaid = array_filter($feePrepaid, static fn(array $item) => $item['sum_'] > 0);
+		}
+
+		foreach ($collection as $statusRow) {
+			foreach ($activePairs as $pair) {
+				if ($statusRow->id === $pair['status']) {
+					$statusRow->status = RewardBonusValidatorInterface::STATUS_DONE_OK;
+					$isFee = false;
+					$hasNotCalls = false;
+
+					foreach ($withOutCalls as $withOutCall) {
+						if ($withOutCall['status'] === $pair['status']) {
+							$hasNotCalls = true;
+						}
+					}
```

Comment:
```suggestion
					foreach ($withOutCalls as $withOutCall) {
						if ($withOutCall['status'] === $pair['status']) {
							$hasNotCalls = true;
							break;
						}
					}
```
иначе цикл идёт по всему массиву, а не до первого совпадения.

Reply:
Поправил